src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -453,10 +483,29 @@ def get_temperature(actual_temperature_lines: list[str] | None) -> float | None:
     )
 
 
+def get_kinetic_measurements(
+    kinetic_lines: list[str] | None,
+) -> tuple[dict[str, list[float | None]], list[float]] | None:
+    if not kinetic_lines:
+        return None
+    data = read_csv(StringIO("\n".join(kinetic_lines)), sep="\t")
+    kinetic_measurements: defaultdict[str, list[float | None]] = defaultdict(
+        list[float | None]
+    )
+    kinetic_elapsed_time: list[float] = []
+    for col in data.columns:
+        for _idx, value in enumerate(data[col]):
+            if col == "Time":
+                kinetic_elapsed_time.append(_convert_time_to_seconds(str(value)))
+            elif not pd.isna(value):
```

Comment:
This isn't the correct way to do this - you are constructing lists from the columns of a dataframe. If you drop NA values in the middle of a column, you will end up with mis-aligned data.

Looking at the sample data, I think you are doing this because there is an entire column with NA values. So what we really want to do is drop columns that are all NA, and then error is a column has some NA and some not. 

Note that you can drop all null columns by calling `.dropna(axis="columns", how="all")` when creating the data